### ISSUE TITLE
SlateJS crashes when clicking selection of the whole text

### ISSUE THREAD

**Issue by timobehrens** [ianstormtaylor/slate#2425]
#### What's the current behavior?

When you select the whole text and then click on the selection, slateJS crashes.

JSFiddle: slatejs demo https://www.slatejs.org/#/rich-text
OS: MacOS 10.13.6
Browser: Firefox 63.0.1

#### What's the expected behavior?

Not to crash. Deselection should happen.

![slate-error](https://user-images.githubusercontent.com/15941188/48547604-55c85e80-e8cb-11e8-886f-d5d3fcc22f62.gif)

This is the error:
```
a/n.renderBlockButton@https://www.slatejs.org/main-1783ad63c9cf9b77dd4d.js:42:22760
value@https://www.slatejs.org/main-1783ad63c9cf9b77dd4d.js:42:25050
Mi@https://www.slatejs.org/main-1783ad63c9cf9b77dd4d.js:22:62666
Ti@https://www.slatejs.org/main-1783ad63c9cf9b77dd4d.js:22:67552
mo@https://www.slatejs.org/main-1783ad63c9cf9b77dd4d.js:22:77129
fo@https://www.slatejs.org/main-1783ad63c9cf9b77dd4d.js:22:77436
Zo@https://www.slatejs.org/main-1783ad63c9cf9b77dd4d.js:22:80984
Vo@https://www.slatejs.org/main-1783ad63c9cf9b77dd4d.js:22:80496
Wo@https://www.slatejs.org/main-1783ad63c9cf9b77dd4d.js:22:80322
Io@https://www.slatejs.org/main-1783ad63c9cf9b77dd4d.js:22:79695
yo@https://www.slatejs.org/main-1783ad63c9cf9b77dd4d.js:22:78927
enqueueSetState@https://www.slatejs.org/main-1783ad63c9cf9b77dd4d.js:22:54463
b.prototype.setState@https://www.slatejs.org/main-1783ad63c9cf9b77dd4d.js:14:1264
a/n.onChange@https://www.slatejs.org/main-1783ad63c9cf9b77dd4d.js:42:23717
onChange@https://www.slatejs.org/main-1783ad63c9cf9b77dd4d.js:1:185776
value@https://www.slatejs.org/main-1783ad63c9cf9b77dd4d.js:1:114892
value/<@https://www.slatejs.org/main-1783ad63c9cf9b77dd4d.js:1:114685
o@https://www.slatejs.org/main-1783ad63c9cf9b77dd4d.js:22:161684
C/<@https://www.slatejs.org/main-1783ad63c9cf9b77dd4d.js:22:161806
u@https://www.slatejs.org/main-1783ad63c9cf9b77dd4d.js:22:164856


    in a
    in a
    in a
    in div
    in Styled(div)
    in div
    in a
    in a
    in a
    in a
```

**Comment by ottis** [comment]
Hitting this too @timobehrens. It triggers slightly different for me though. I select the text, remove focus from the text area, then give it back focus (seen in gif below)

JSFiddle: slatejs demo https://www.slatejs.org/#/rich-text
OS: MacOS 10.13.6
Browser: Google Chrome 70.0.3538.102 (Official Build) (64-bit)

![issue](https://user-images.githubusercontent.com/966901/48557915-db560980-e8df-11e8-8d71-d51afcd3b33a.gif)

**Comment by ericedem** [comment]
Looks like this was fixed in #2412

**Comment by brendancarney** [comment]
This appears to be fixed, closing. Please comment if you believe the issue is not resolved. Thanks!
